src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java

```diff
@@ -236,39 +231,18 @@ private ReplicationController createReplicationController(
 	}
 
 	private PodSpec createPodSpec(String appId, AppDeploymentRequest request, Integer port, Integer instanceIndex) {
-		PodSpecBuilder podSpec = new PodSpecBuilder();
 
-		// Add image secrets if set
-		if (properties.getImagePullSecret() != null) {
-			podSpec.addNewImagePullSecret(properties.getImagePullSecret());
-		}
+		PodSpecBuilder podSpec = createPodSpecBuilder(properties);
 
 		Container container = containerFactory.create(appId, request, port, instanceIndex);
-
-		// add memory and cpu resource limits
-		ResourceRequirements req = new ResourceRequirements();
-		req.setLimits(deduceResourceLimits(properties, request));
-		req.setRequests(deduceResourceRequests(properties, request));
-		container.setResources(req);
-		ImagePullPolicy pullPolicy = deduceImagePullPolicy(properties, request);
-		container.setImagePullPolicy(pullPolicy.name());
-		// Add volumes and mounts
+		addMemoryCpu(request, container, properties);
 		if (properties.getHostVolumeMounts() != null) {
-			container.setVolumeMounts(properties.getHostVolumeMounts().stream()
-					.map(hvm -> new VolumeMount(hvm.getContainerPath(), hvm.getName(), hvm.isReadOnly(), null))
-					.collect(Collectors.toList()));
-			podSpec.withVolumes(properties.getHostVolumeMounts().stream()
-					.map(mv -> {
-						Volume volume = new Volume();
-						HostPathVolumeSource hostPath = new HostPathVolumeSource();
-						hostPath.setPath(mv.getHostPath());
-						volume.setHostPath(hostPath);
-						volume.setName(mv.getName());
-						return volume;
-					}).collect(Collectors.toList()));
+			addVolumes(podSpec, properties);
```

Comment:
Can the following:
```
PodSpecBuilder podSpec = createPodSpecBuilder(properties);
addVolumes(podSpec, properties);
```
be moved after the creation of container and other steps (add resource limits, add volume mounts) involving the container?